src/MySqlConnector/MySqlDecimal.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Globalization;
+
+namespace MySqlConnector;
+
+	public struct MySqlDecimal : IComparable, IComparable<MySqlDecimal>, IConvertible, IEquatable<MySqlDecimal>
+	{
+		private readonly string value;
+
+		internal MySqlDecimal(string val)
+		{
+			if (val.Length > 65)
```

Comment:
I _think_ this is incorrect, because 65 is the number of digits, not including the `.` and optional `-` character. (Would need to double-check though.)

Reply:
I'd like to see additional validation here, such as `/^-?([1-9][0-9]*|0)(\.[0-9]+)?$/`.

Reply:
Reading from  [here](https://dev.mysql.com/doc/refman/8.0/en/fixed-point-types.html) and [here](https://dev.mysql.com/doc/refman/8.0/en/precision-math-decimal-characteristics.html)

The declaration syntax for a DECIMAL column is DECIMAL(M,D). The ranges of values for the arguments are as follows:

M is the maximum number of digits (the precision). It has a range of 1 to 65.
D is the number of digits to the right of the decimal point (the scale). It has a range of 0 to 30 and must be no larger than M.
DECIMAL columns do not store a leading + character or - character or leading 0 digits. If you insert +0003.1 into a DECIMAL(5,1) column, it is stored as 3.1. For negative numbers, a literal - character is not stored.

Not sure how _-999.99_ which can be seen in mysql table on mysql workbench is stored and how it will be converted to a string and then converted back.

Reply:
I am removing the length check for now just checking format using regex.

Reply:
@bgrainger 

Turns out `/^-?([1-9][0-9]*|0)(\.[0-9]+)?$/` should have been `/^-?([0-9][0-9]*|0)(\.[0-9]+)?$/` to support zero fill decimals.

Will submit a PR for this later on at some point, just wanted to comment on the reference now that I've found it.

Reply:
FWIW `[0-9][0-9]*` is equivalent to `[0-9]+` and then the trailing `|0` is superfluous.

Reply:
https://github.com/mysql-net/MySqlConnector/issues/1354